src/app/pages/reportsdashboard/components/report/report.component.html

```diff
@@ -153,6 +154,22 @@ <h4>
             </table>
 
             </div>
+            <ng-container *ngIf="report.stacked_show_total">
+
+              <div *ngIf="report && data && data.legend" class="legend-timestamp" fxFlex="32px">
```

Comment:
it seems the check for `report` is unnecessary because it should already exist based on the expression above `*ngIf="report.stacked_show_total"`. Like @undsoft mentioned earlier, we can simplify it into `data?.legend`